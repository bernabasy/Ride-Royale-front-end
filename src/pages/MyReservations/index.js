import React, { useEffect, useState } from 'react';
import axios from 'axios';
import Layout from '../../components/Layout/Layout';

const MyReservationsPage = () => (

const MyReservations = () => {
  const [reservations, setReservations] = useState([]);
  const [loading, setLoading] = useState(false);
  const user = JSON.parse(localStorage.getItem('user'));

  useEffect(() => {
    setLoading(true);
    axios
      .get(
        `http://127.0.0.1:3000/api/v1/users/${user.user.id}/reservations`,
      )
      .then((response) => {
        setReservations(response.data.reservations);
        setLoading(false);
      });
  }, []);
  if (reservations.length <= 0) {
    if (loading) {
      return (
        <div className="container-fluid d-flex justify-content-center align-items-center vh-100">
          <i className="fa-solid fa-spinner fa-spin fs-1" />
        </div>
      );
    }
  }

  return (
    <Layout>
      <div className="min-vh-100 container page-wrapper">
        <h4 className="text-center my-4">My Reservations</h4>

        <div className="row">
          {
            reservations.map((reservation) => (
              <ReservationCardItem
                key={reservation.reservation.id}
                data={reservation}
              />
            ))
          }
        </div>
      </div>
    </Layout>
  );
};

export default MyReservationsPage;
