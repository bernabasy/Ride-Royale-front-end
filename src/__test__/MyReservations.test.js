import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import MyReservations from '../pages/MyReservations';

describe('MyReservations tests', () => {
  it('renders correctly', () => {
    const tree = render(
      <BrowserRouter>
        <MyReservations />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
