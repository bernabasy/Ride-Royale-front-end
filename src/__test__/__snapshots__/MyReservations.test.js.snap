// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyReservations tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="row gx-0"
      >
        <nav
          class="container-fluid navbar navbar-light bg-light shadow-sm status-bar"
        >
          <div
            class="row d-flex align-items-center justify-content-start"
          >
            <div
              class="col"
            >
              <button
                class="btn"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="font-size: 1.5rem;"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </nav>
        <aside
          class="col-md-2 border-end bg-white vh-100 aside-hidden sidebar"
        >
          <button
            class="btn float-end aside-close"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="font-size: 1.5rem;"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
              />
            </svg>
          </button>
          <h4
            class="fw-bold text-center my-4"
          >
            Ride Royale
          </h4>
          <ul
            class="list-group rounded-0 mt-4"
          >
            <li
              class="mx-2 list-group-item m-0 p-0 border-0 my-2 side-nav-item"
            >
              <a
                class="({
      isActive
    }) => isActive ? _styleModule.default['active-link'] : _styleModule.default['not-active-link'] text-decoration-none side-nav-link"
                href="/cars"
              >
                <svg
                  class="icon"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
                <span
                  class="side-nav-span"
                >
                  Home
                </span>
              </a>
            </li>
            <li
              class="mx-2 list-group-item m-0 p-0 border-0 my-2 side-nav-item"
            >
              <a
                class="({
      isActive
    }) => isActive ? _styleModule.default['active-link'] : _styleModule.default['not-active-link'] text-decoration-none side-nav-link"
                href="/new-reservation"
              >
                <svg
                  class="icon"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
                  />
                </svg>
                <span
                  class="side-nav-span"
                >
                  New Reservation
                </span>
              </a>
            </li>
            <li
              class="mx-2 list-group-item m-0 p-0 border-0 my-2 side-nav-item"
            >
              <a
                class="({
      isActive
    }) => isActive ? _styleModule.default['active-link'] : _styleModule.default['not-active-link'] text-decoration-none side-nav-link"
                href="/reservations"
              >
                <svg
                  class="icon"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V9h14v10zM5 7V5h14v2H5zm2 4h10v2H7zm0 4h7v2H7z"
                  />
                </svg>
                <span
                  class="side-nav-span"
                >
                  My Reservation
                </span>
              </a>
            </li>
            <li
              class="mx-2 list-group-item m-0 p-0 border-0 my-2 side-nav-item"
            >
              <a
                class="({
      isActive
    }) => isActive ? _styleModule.default['active-link'] : _styleModule.default['not-active-link'] text-decoration-none side-nav-link"
                href="/add-car"
              >
                <svg
                  class="icon"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                <span
                  class="side-nav-span"
                >
                  Add Car
                </span>
              </a>
            </li>
            <li
              class="mx-2 list-group-item m-0 p-0 border-0 my-2 side-nav-item"
            >
              <a
                class="({
      isActive
    }) => isActive ? _styleModule.default['active-link'] : _styleModule.default['not-active-link'] text-decoration-none side-nav-link"
                href="/delete-car"
              >
                <svg
                  class="icon"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M16 9v10H8V9h8m-1.5-6h-5l-1 1H5v2h14V4h-3.5l-1-1zM18 7H6v12c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7z"
                  />
                </svg>
                <span
                  class="side-nav-span"
                >
                  Delete Car
                </span>
              </a>
            </li>
          </ul>
        </aside>
        <div
          class="col-md-10 main-wrapper"
        >
          <h1>
            My Reservations
          </h1>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="row gx-0"
    >
      <nav
        class="container-fluid navbar navbar-light bg-light shadow-sm status-bar"
      >
        <div
          class="row d-flex align-items-center justify-content-start"
        >
          <div
            class="col"
          >
            <button
              class="btn"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="font-size: 1.5rem;"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                />
              </svg>
            </button>
          </div>
        </div>
      </nav>
      <aside
        class="col-md-2 border-end bg-white vh-100 aside-hidden sidebar"
      >
        <button
          class="btn float-end aside-close"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="font-size: 1.5rem;"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
            />
          </svg>
        </button>
        <h4
          class="fw-bold text-center my-4"
        >
          Ride Royale
        </h4>
        <ul
          class="list-group rounded-0 mt-4"
        >
          <li
            class="mx-2 list-group-item m-0 p-0 border-0 my-2 side-nav-item"
          >
            <a
              class="({
      isActive
    }) => isActive ? _styleModule.default['active-link'] : _styleModule.default['not-active-link'] text-decoration-none side-nav-link"
              href="/cars"
            >
              <svg
                class="icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <span
                class="side-nav-span"
              >
                Home
              </span>
            </a>
          </li>
          <li
            class="mx-2 list-group-item m-0 p-0 border-0 my-2 side-nav-item"
          >
            <a
              class="({
      isActive
    }) => isActive ? _styleModule.default['active-link'] : _styleModule.default['not-active-link'] text-decoration-none side-nav-link"
              href="/new-reservation"
            >
              <svg
                class="icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
                />
              </svg>
              <span
                class="side-nav-span"
              >
                New Reservation
              </span>
            </a>
          </li>
          <li
            class="mx-2 list-group-item m-0 p-0 border-0 my-2 side-nav-item"
          >
            <a
              class="({
      isActive
    }) => isActive ? _styleModule.default['active-link'] : _styleModule.default['not-active-link'] text-decoration-none side-nav-link"
              href="/reservations"
            >
              <svg
                class="icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V9h14v10zM5 7V5h14v2H5zm2 4h10v2H7zm0 4h7v2H7z"
                />
              </svg>
              <span
                class="side-nav-span"
              >
                My Reservation
              </span>
            </a>
          </li>
          <li
            class="mx-2 list-group-item m-0 p-0 border-0 my-2 side-nav-item"
          >
            <a
              class="({
      isActive
    }) => isActive ? _styleModule.default['active-link'] : _styleModule.default['not-active-link'] text-decoration-none side-nav-link"
              href="/add-car"
            >
              <svg
                class="icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
              <span
                class="side-nav-span"
              >
                Add Car
              </span>
            </a>
          </li>
          <li
            class="mx-2 list-group-item m-0 p-0 border-0 my-2 side-nav-item"
          >
            <a
              class="({
      isActive
    }) => isActive ? _styleModule.default['active-link'] : _styleModule.default['not-active-link'] text-decoration-none side-nav-link"
              href="/delete-car"
            >
              <svg
                class="icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M16 9v10H8V9h8m-1.5-6h-5l-1 1H5v2h14V4h-3.5l-1-1zM18 7H6v12c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7z"
                />
              </svg>
              <span
                class="side-nav-span"
              >
                Delete Car
              </span>
            </a>
          </li>
        </ul>
      </aside>
      <div
        class="col-md-10 main-wrapper"
      >
        <h1>
          My Reservations
        </h1>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
